tests/metal/simple-task.slang

```diff
@@ -1,6 +1,11 @@
-//TEST:SIMPLE(filecheck=CHECK): -target metal
-//TEST:SIMPLE(filecheck=CHECK-ASM): -target metallib
-//TEST:REFLECTION(filecheck=REFLECT):-target metal -entry main_kernel -stage task
+// TEST:SIMPLE(filecheck=CHECK): -target metal
+// TEST:SIMPLE(filecheck=CHECK-ASM): -target metallib
+// TEST:REFLECTION(filecheck=REFLECT):-target metal -entry main_kernel -stage task
+
+// CHECK:      0
```

Comment:
I don't quite understand these results. Can we put something more meaningful here, such as:
```
// CHECK: _slang_mgp.set_threadgroups_per_grid
```

Reply:
I added some checks that should work, however I can't get the tests to run on my macbook, the runner keeps ignoring them.

Reply:
Our test runner just try to run the `metal` command directly, so it needs to be made available from PATH. We should probably enhance the logic in `locateMetalCompiler` in `slang-metal-compiler.cpp` so that it can discover metal sdk on macos.